src/server.rs

```diff
@@ -296,6 +301,20 @@ impl LanguageServer for LspServer {
         })
     }
     async fn shutdown(&self) -> RpcResult<()> {
+        let mut client = match self.client.lock() {
+            Ok(client) => client,
+            Err(err) => {
+                return RpcResult::Err(RpcError {
+                    code: RpcErrorCode::InternalError,
+                    message: format!("{}", err),
+                    data: None,
+                })
+            }
+        };
+        // Replace the original client with None causing the original to be dropped.
```

Comment:
I think we should add a `XXX` comment with a link to the bug and PR so we have an indicator for when that can go away.